File: problems/p/pauleigon/pauleigon.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <iostream>

using namespace std;

int main() {
    int n, p, q, quocient;

    cin >> n >> p >> q;

    quocient = (p + q) / n;
    if (quocient % 2 == 0) {
        cout << "paul" << endl;
    }
    else {
        cout << "opponent" << endl;
    }
}

File: problems/p/pauleigon/pauleigon.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    n, p, q = [int(val) for val in input().split()]

    quocient = (p + q) // n
    if quocient % 2 == 0:
        print("paul")
    else:
        print("opponent")


if __name__ == '__main__':
    main()